### PR TITLE
refactor(story): page article bottom

### PR DESCRIPTION
**Pull request by LinHeMa**
### info
[PRD](https://paper.dropbox.com/doc/Mirror-TV-Article-Page-story-0ksuf6l2ubyEPRYZdiEgP)
[Figma](https://www.figma.com/design/BATB1ux8KjfWYa4DV1Xm3t/2024_%E9%8F%A1%E9%9B%BB%E8%A6%96_Web?node-id=0-2487&node-type=canvas&t=j4kwWyg96VjCjDUM-0)

## tasks
- migration 更多新聞、追蹤我們components

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208332170527656

### CODE REVIEW

File: packages/mirror-tv-next/app/story/[slug]/layout.tsx
Comment (erase2004) [inline packages/mirror-tv-next/app/story/[slug]/layout.tsx:1]: `ArticleRelatedPost`s 和 `ArticleSocilaList` 應該放到 page 去處理

File: packages/mirror-tv-next/app/_actions/story/fetch-story-post-by-slug.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      variables: {
+        slug,
+      },
+    })
+    return data.allPosts ?? []
+  } catch (err) {
+    const annotatingError = errors.helpers.wrap(
+      err,
+      'UnhandledError',
+      'Error occurs while fetching category data in category page'
```
Comment (erase2004) [inline packages/mirror-tv-next/app/_actions/story/fetch-story-post-by-slug.ts]: 錯誤訊息不正確

File: packages/mirror-tv-next/styles/theme/_variable.scss [hunk 2]
```diff
@@ -8,6 +8,10 @@ $header-max-width-xl: 1000px;
 $header-max-width-xxl: 1200px;
 $header-padding-md: 40px;
 
+$story-max-width-md: 688px;
+$story-wrapper-max-width-md: 600px;
+$story-max-width-xl: 1080px;
```
Comment (erase2004) [inline packages/mirror-tv-next/styles/theme/_variable.scss:13]: 這個目前沒有用到？

File: packages/mirror-tv-next/components/story/article-related-posts.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
```
Comment (erase2004) [inline packages/mirror-tv-next/components/story/article-related-posts.tsx]: https://github.com/mirror-tv/Tabris/pull/218#issuecomment-2537918700

File: packages/mirror-tv-next/app/story/[slug]/_styles/story.module.scss [hunk 4]
```diff
@@ -17,3 +32,39 @@
...
+  @include media-breakpoint-up(xl) {
+    max-width: $max-width-xl;
+    display: flex;
+  }
+  @include media-breakpoint-up(xxl) {
+    max-width: $max-width-xxl;
+  }
+}
+
+.microAd {
```
Comment (erase2004) [inline packages/mirror-tv-next/app/story/[slug]/_styles/story.module.scss]: `.microAd` 似乎尚未被使用

File: packages/mirror-tv-next/constants/social-medial.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
+interface SocialMediaConfig {
+  [type: string]: {
+    href: string
+    image: string
+  }
+}
+
+const socialMediaConfig: SocialMediaConfig = {
```
Comment (erase2004) [inline packages/mirror-tv-next/constants/social-medial.ts]: 建議用 `as const satisfies SocialMediaConfig`

File: packages/mirror-tv-next/constants/social-medial.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    href: 'https://lin.ee/4XsO8xi',
+    image: '/images/social-media/line-round.svg',
+  },
+  twitter: {
+    href: 'https://twitter.com/mnews_tw',
+    image: '/images/social-media/twitter-round.svg',
+  },
+}
+
+const socialMediaOrder = ['facebook', 'instagram', 'youtube', 'line', 'twitter']
```
Comment (erase2004) [inline packages/mirror-tv-next/constants/social-medial.ts]: 建議用 `as const`

File: packages/mirror-tv-next/components/story/article-social-list.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+const ArticleSocilaList = () => {
+  return (
+    <div className={styles.followUsWrapper}>
+      <p className={styles.followUs}>追蹤我們</p>
+      <br className={styles.desktopOnly} />
+      <ul className={styles.socialMediaList}>
+        {socialMediaOrder.map((socialMedia) => {
+          const socialMediaData = socialMediaConfig[socialMedia]
+          const imageSize = 50
+          const defaultImageSrc = '/images/image-default.jpg'
```
Comment (erase2004) [inline packages/mirror-tv-next/components/story/article-social-list.tsx:14]: 這個應該不用

File: packages/mirror-tv-next/components/story/article-social-list.tsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+          if (!socialMediaData) {
```
Comment (erase2004) [inline packages/mirror-tv-next/components/story/article-social-list.tsx]: 這一區塊應該也用

File: packages/mirror-tv-next/components/story/article-social-list.tsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+            )
+          }
+
+          const { image, href } = socialMediaData
+          const imageConfig = {
+            alt: socialMedia,
+            src: image || defaultImageSrc,
+            width: imageSize,
+            height: imageSize,
+          }
```
Comment (erase2004) [inline packages/mirror-tv-next/components/story/article-social-list.tsx]: 建議加上 `satisfies Parameters<typeof Image>[0]`

File: packages/mirror-tv-next/components/story/article-social-list.tsx [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+          return (
+            <li key={socialMedia}>
+              <Link href={href}>
```
Comment (erase2004) [inline packages/mirror-tv-next/components/story/article-social-list.tsx]: 這些外部連結應該用新分頁開啟？另外，因為是外部連結，其實用 `next/link` 效益不大，建議直接用 `<a />` 處理

File: packages/mirror-tv-next/graphql/query/story.ts [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+  relatedPosts: RelatedPost[]
+  download: Download[]
+}
+
+export interface FetchStoryBySlugResponse {
+  allPosts: Post[]
+}
+
+const fetchStoryBySlug = gql`
+  query fetchStoryBySlug($slug: String!) {
```
Comment (erase2004) [inline packages/mirror-tv-next/graphql/query/story.ts:17]: 還沒使用到的 field 建議先移除

File: packages/mirror-tv-next/app/story/[slug]/layout.tsx
Comment (LinHeMa) [reply packages/mirror-tv-next/app/story/[slug]/layout.tsx:1]: fixed at [b6f67b2](https://github.com/mirror-tv/Tabris/pull/220/commits/b6f67b2826335acf136348a1845b23cdf358fe19)

Comment (LinHeMa) [reply packages/mirror-tv-next/app/story/[slug]/layout.tsx:1]: fixed at [b6f67b2](https://github.com/mirror-tv/Tabris/pull/220/commits/b6f67b2826335acf136348a1845b23cdf358fe19)

File: packages/mirror-tv-next/app/_actions/story/fetch-story-post-by-slug.ts (hunk 1 above)
Comment (LinHeMa) [reply packages/mirror-tv-next/app/_actions/story/fetch-story-post-by-slug.ts]: fixed at [f0dba9b](https://github.com/mirror-tv/Tabris/pull/220/commits/f0dba9b0f2ef887f24149965e316de3451bcc2ae)

File: packages/mirror-tv-next/styles/theme/_variable.scss (hunk 2 above)
Comment (LinHeMa) [reply packages/mirror-tv-next/styles/theme/_variable.scss:13]: fixed at [02faac2](https://github.com/mirror-tv/Tabris/pull/220/commits/02faac27679a48685f5e617445d19cd01c83f4d1)

File: packages/mirror-tv-next/components/story/article-related-posts.tsx (hunk 3 above)
Comment (LinHeMa) [reply packages/mirror-tv-next/components/story/article-related-posts.tsx]: 感謝提醒 [9c43599](https://github.com/mirror-tv/Tabris/pull/220/commits/9c43599bdc572df74d666c9d3d78699f5c526e82)

File: packages/mirror-tv-next/app/story/[slug]/_styles/story.module.scss (hunk 4 above)
Comment (LinHeMa) [reply packages/mirror-tv-next/app/story/[slug]/_styles/story.module.scss]: 清除 [6d7b2fb](https://github.com/mirror-tv/Tabris/pull/220/commits/6d7b2fbb53a9f9aab454d9841f4022694773c60d)

File: packages/mirror-tv-next/constants/social-medial.ts (hunk 5 above)
Comment (LinHeMa) [reply packages/mirror-tv-next/constants/social-medial.ts]: 感謝建議 [d88a9f2](https://github.com/mirror-tv/Tabris/pull/220/commits/d88a9f21b0659720b0d8edcc3675ea3950097055)

File: packages/mirror-tv-next/constants/social-medial.ts (hunk 6 above)
Comment (LinHeMa) [reply packages/mirror-tv-next/constants/social-medial.ts]: 感謝建議 [d88a9f2](https://github.com/mirror-tv/Tabris/pull/220/commits/d88a9f21b0659720b0d8edcc3675ea3950097055)

File: packages/mirror-tv-next/components/story/article-social-list.tsx (hunk 8 above)
Comment (LinHeMa) [reply packages/mirror-tv-next/components/story/article-social-list.tsx]: [81ee922](https://github.com/mirror-tv/Tabris/pull/220/commits/81ee9224cac52e6b2f94eaeed2c153daffd5feea)
如果這樣做呢？

File: packages/mirror-tv-next/graphql/query/story.ts (hunk 11 above)
Comment (LinHeMa) [reply packages/mirror-tv-next/graphql/query/story.ts:17]: 修復 [587d4bb](https://github.com/mirror-tv/Tabris/pull/220/commits/587d4bb048337c60851ad4e9d0062933dcde0f99)

File: packages/mirror-tv-next/app/_actions/story/fetch-story-post-by-slug.ts [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+      variables: {
+        slug,
+      },
+    })
+    return data.allPosts ?? []
+  } catch (err) {
+    const annotatingError = errors.helpers.wrap(
+      err,
+      'UnhandledError',
+      'Error occurs while fetching slug data in story page'
```
Comment (erase2004) [inline packages/mirror-tv-next/app/_actions/story/fetch-story-post-by-slug.ts]: 建議加上 slug 資訊，增加辨識度

File: packages/mirror-tv-next/app/story/[slug]/layout.tsx [hunk 13]
```diff
@@ -1,15 +1,21 @@
 import Aside from '~/components/story/aside'
 import styles from './_styles/story.module.scss'
+import { cloneElement } from 'react'
 
-export default function StoryPageLayout({
+export default async function StoryPageLayout({
```
Comment (erase2004) [inline packages/mirror-tv-next/app/story/[slug]/layout.tsx]: async 應該不用

File: packages/mirror-tv-next/app/story/[slug]/layout.tsx [hunk 14]
```diff
@@ -1,15 +1,21 @@
 import Aside from '~/components/story/aside'
 import styles from './_styles/story.module.scss'
+import { cloneElement } from 'react'
```
Comment (erase2004) [inline packages/mirror-tv-next/app/story/[slug]/layout.tsx]: cloneElement 應該不用？

File: packages/mirror-tv-next/constants/social-medial.ts [hunk 15]
```diff
@@ -0,0 +1,39 @@
+interface SocialMediaConfig {
+  [type: string]: {
+    href: string
+    image: string
+  }
+}
+
+const socialMediaConfig: SocialMediaConfig = {
```
Comment (erase2004) [inline packages/mirror-tv-next/constants/social-medial.ts]: `socialMediaConfig: SocialMediaConfig` 這個就不用保留了

File: packages/mirror-tv-next/graphql/query/story.ts [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+    allPosts(where: { slug: $slug, state_not_in: invisible }) {
+      relatedPosts(where: { state: published }) {
+        slug
+        name
+        heroImage {
```
Comment (erase2004) [inline packages/mirror-tv-next/graphql/query/story.ts]: heroImage 有用到嗎？

Comment (LinHeMa) [reply packages/mirror-tv-next/graphql/query/story.ts]: fixed at [66e0307](https://github.com/mirror-tv/Tabris/pull/220/commits/66e03073023947cf2b9a7ca29dc8b4a6b6d668c5)

File: packages/mirror-tv-next/constants/social-medial.ts (hunk 15 above)
Comment (LinHeMa) [reply packages/mirror-tv-next/constants/social-medial.ts]: fixed at [5f94639](https://github.com/mirror-tv/Tabris/pull/220/commits/5f946396ff48e1212d66d0762d8a8da1e8c271bd)

File: packages/mirror-tv-next/app/story/[slug]/layout.tsx (hunk 14 above)
Comment (LinHeMa) [reply packages/mirror-tv-next/app/story/[slug]/layout.tsx]: fixed at [cfacccb](https://github.com/mirror-tv/Tabris/pull/220/commits/cfacccb480bdec51874cadd4d5d2c2847c88134a)

File: packages/mirror-tv-next/app/story/[slug]/layout.tsx (hunk 13 above)
Comment (LinHeMa) [reply packages/mirror-tv-next/app/story/[slug]/layout.tsx]: fixed at [48bceaf](https://github.com/mirror-tv/Tabris/pull/220/commits/48bceaf43e1d2dbaec24aa069d19ba3d5e8621a8)

File: packages/mirror-tv-next/app/_actions/story/fetch-story-post-by-slug.ts (hunk 12 above)
Comment (LinHeMa) [reply packages/mirror-tv-next/app/_actions/story/fetch-story-post-by-slug.ts]: 感謝提醒 [7ce95b1](https://github.com/mirror-tv/Tabris/pull/220/commits/7ce95b196e8cd547c53100c698831e14de0efc45)

File: packages/mirror-tv-next/graphql/query/story.ts [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+export interface Tag {
+  name: string
+}
+
+export interface RelatedPost {
+  slug: string
+  name: string
+  heroImage: {
```
Comment (erase2004) [inline packages/mirror-tv-next/graphql/query/story.ts]: type 資訊請一併調整

Comment (LinHeMa) [reply packages/mirror-tv-next/graphql/query/story.ts]: fixed at [b265977](https://github.com/mirror-tv/Tabris/pull/220/commits/b2659778fc7b49f8294bec59a364a8113e57837b)

File: packages/mirror-tv-next/components/story/article-related-posts.tsx [hunk 18]
```diff
@@ -0,0 +1,23 @@
+import UiHeadingBordered from '../shared/ui-heading-bordered'
+import { Post } from '~/graphql/query/story'
```
Comment (erase2004) [inline packages/mirror-tv-next/components/story/article-related-posts.tsx]: 改名後壞了

Comment (LinHeMa) [reply packages/mirror-tv-next/components/story/article-related-posts.tsx]: 修正 [7db674d](https://github.com/mirror-tv/Tabris/pull/220/commits/7db674d98c24235ceac695629d071a2dd3cca660)

**Review by erase2004 — APPROVED** [review]
Accepted ✔️

**Review by v61265 — APPROVED** [review]
LGTM～讚讚～

**Pull request merged**